src/rust/aggregator/src/pipelines/fees.rs

```diff
@@ -0,0 +1,60 @@
+use chrono::{DateTime, Duration, Utc};
+use sqlx::PgPool;
+
+use aggregator::{util::*, Pipeline, PipelineAggregationResult};
+use sqlx_postgres::PgConnection;
+
+pub const TIMEOUT: std::time::Duration = std::time::Duration::from_secs(60);
+
+pub struct Fees {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl Fees {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Pipeline for Fees {
+    fn model_name(&self) -> String {
+        String::from("Fees")
+    }
+
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap() + Duration::from_std(TIMEOUT).unwrap()
+                < Utc::now()
+    }
+
+    async fn process_and_save_historical_data(&mut self) -> PipelineAggregationResult {
+        self.process_and_save_internal().await
+    }
+
+    fn poll_interval(&self) -> Option<std::time::Duration> {
+        Some(TIMEOUT)
+    }
+
+    async fn process_and_save_internal(&mut self) -> PipelineAggregationResult {
+        let mut transaction = create_repeatable_read_transaction(&self.pool).await?;
+        sqlx::query_file!("sqlx_queries/fees/backfill.sql",)
+            .execute(&mut transaction as &mut PgConnection)
+            .await
+            .map_err(to_pipeline_error)?;
+        sqlx::query_file!("sqlx_queries/fees/delete_last_indexed_txn.sql",)
+            .execute(&mut transaction as &mut PgConnection)
+            .await
+            .map_err(to_pipeline_error)?;
+        sqlx::query_file!("sqlx_queries/fees/update_last_indexed_txn.sql",)
```

Comment:
How about just using a simple SQL statement to cover these two? E.g. one that tries to insert, and overwrites on conflict?

Reply:
I may have miscommunicated above, because I'm confused by your response. Perhaps what I mean was:

How about combining `delete_last_indexed_txn.sql` and `update_last_indexed_txn.sql` into a single query?

Not critical though, and the functionality is still assured by the current logic so I'll mark this as resolved

Reply:
Per offline discussion, noting that the intended consolidation could be achieved via an `UPDATE` statement per `user_history/set_new_last_indexed_txn_version.sql`, but is not required